pkg/xray/watches.go

```diff
@@ -0,0 +1,246 @@
+			r = append(r, unpackProjectResource(res))
+		}
+		pr.Resources = r
+	}
+	watch.ProjectResources = pr
+
+	ap := []WatchAssignedPolicy{}
+	if v, ok := d.GetOk("assigned_policy"); ok {
+		for _, pol := range v.([]interface{}) {
+			ap = append(ap, unpackAssignedPolicy(pol))
+		}
+	}
+	watch.AssignedPolicies = ap
+
+	return watch
+}
+
+func unpackProjectResource(rawCfg interface{}) WatchProjectResource {
+	resource := new(WatchProjectResource)
+
+	cfg := rawCfg.(map[string]interface{})
+	resource.Type = cfg["type"].(string)
+
+	if v, ok := cfg["bin_mgr_id"]; ok {
+		resource.BinaryManagerId = v.(string)
+	}
+	if v, ok := cfg["name"]; ok {
+		resource.Name = v.(string)
+	}
+
+	if v, ok := cfg["filter"]; ok {
+		resourceFilters := unpackFilters(v.([]interface{}))
+		resource.Filters = resourceFilters
+	}
+
+	return *resource
+}
+
+func unpackFilters(list []interface{}) []WatchFilter {
+	var filters []WatchFilter
+
+	for _, raw := range list {
+		filter := new(WatchFilter)
+		f := raw.(map[string]interface{})
+		filter.Type = f["type"].(string) // TODO: recognize the type of the filter
+		filter.Value = f["value"].(string)
+		filters = append(filters, *filter)
+	}
+
+	return filters
+}
+
+func unpackAssignedPolicy(rawCfg interface{}) WatchAssignedPolicy {
+	policy := new(WatchAssignedPolicy)
+
+	cfg := rawCfg.(map[string]interface{})
+	policy.Name = cfg["name"].(string)
+	policy.Type = cfg["type"].(string)
+
+	return *policy
+}
+
+func packProjectResources(resources WatchProjectResources) []interface{} {
+	if len(resources.Resources) == 0 {
+		return []interface{}{}
+	}
+
+	var list []interface{}
+	for _, res := range resources.Resources {
+		resourceMap := map[string]interface{}{}
+		resourceMap["type"] = res.Type
+		if len(res.Name) > 0 {
+			resourceMap["name"] = res.Name
+		}
+		if len(res.BinaryManagerId) > 0 {
+			resourceMap["bin_mgr_id"] = res.BinaryManagerId
+		}
+		resourceMap["filter"] = packFilters(res.Filters)
+		list = append(list, resourceMap)
+	}
+
+	return list
+}
+
+func packFilters(filters []WatchFilter) []interface{} {
+	if filters == nil {
+		return []interface{}{}
+	}
+
+	var l []interface{}
+	for _, f := range filters {
+		m := map[string]interface{}{
+			"type":  f.Type,
+			"value": f.Value,
+		}
+		l = append(l, m)
+	}
+
+	return l
+}
+
+func packAssignedPolicies(policies []WatchAssignedPolicy) []interface{} {
+	if policies == nil {
+		return []interface{}{}
+	}
+
+	var l []interface{}
+	for _, p := range policies {
+		m := make(map[string]interface{})
+		m["name"] = p.Name
+		m["type"] = p.Type
+		l = append(l, m)
+	}
+
+	return l
+}
+
+func resourceXrayWatchCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	watch := unpackWatch(d)
+	_, err := m.(*resty.Client).R().SetBody(watch).Post("xray/api/v2/watches")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(watch.GeneralData.Name) // ID may be returned according to the API docs, but not in go-xray
```

Comment:
What does the comment mean?

Reply:
ops, leftover from the old code. Removed